dandi/move.py

```diff
@@ -812,6 +812,7 @@ def move(
                 raise TypeError("`dandiset` must be a Path when work_on='both'")
             local_ds, subpath = find_dandiset_and_subpath(dandiset)
             client = DandiAPIClient.for_dandi_instance(dandi_instance)
+            client.dandi_authenticate()
```

Comment:
in principle we do not need to authenticate if doing `dry_run` do we?

Would it be feasible in general to provide logic to authenticate when receiving 401 and haven't been authenticated before?